k8s.io/kubernetes/cmd/kubeadm/app/cmd/phases/workflow TestBindToCommandArgRequirements fails in Go 1.21

### What happened?

`FORCE_HOST_GO=y make test WHAT=./cmd/kubeadm/app/cmd/phases/workflow` fails with the latest Go 1.21 dev branch, with:

```
+++ [0520 15:18:01] Setting GOMAXPROCS: 48
+++ [0520 15:18:01] Running tests without code coverage and with -race
--- FAIL: TestBindToCommandArgRequirements (0.00s)
    --- FAIL: TestBindToCommandArgRequirements/leaf_command,_no_defined_args,_follow_parent (0.00s)
        runner_test.go:406: The function pointers where not equal.
    --- FAIL: TestBindToCommandArgRequirements/container_cmd_expect_none,_custom_arg_check_for_leaf (0.00s)
        runner_test.go:406: The function pointers where not equal.
FAIL
FAIL	k8s.io/kubernetes/cmd/kubeadm/app/cmd/phases/workflow	0.016s
FAIL
```



### What did you expect to happen?

Test should pass but it fails.

### How can we reproduce it (as minimally and precisely as possible)?

Build the latest Go tree from source (or wait for Go 1.21 rc1). Then run:

    FORCE_HOST_GO=y make test WHAT=./cmd/kubeadm/app/cmd/phases/workflow

### Anything else we need to know?

This test is in `cmd/kubeadm/app/cmd/phases/workflow/runner_test.go`.

It is comparing function pointers by using reflect on the function value and then checking .Pointer. This typically does work for ordinary functions, but the test is comparing closures instead. If the closures were created by calling the same function twice, they may be different closures. This is why Go does not support function equality directly at all (which is why the test is resorting to reflect and non-guaranteed behavior).

The closures appear to come from, among other places, cobra.ExactArgs:

```
% go doc -src cobra.ExactArgs
package cobra // import "github.com/spf13/cobra"

// ExactArgs returns an error if there are not exactly n args.
func ExactArgs(n int) PositionalArgs {
	return func(cmd *Command, args []string) error {
		if len(args) != n {
			return fmt.Errorf("accepts %d arg(s), received %d", n, len(args))
		}
		return nil
	}
}
% 
```

The Kubernetes test itself needs to be rewritten not to depend on function pointer equality. In fact, as the comment notes, in the earlier Go implementation, checking .Pointer made it appear that ExactArgs(1) == ExactArgs(3), because Pointer only looks at the code pointer, not the associated data pointer:

```
	// because cobra.ExactArgs(1) == cobra.ExactArgs(3), it is needed
	// to run test argument sets that both pass and fail to ensure the correct function was set.
```

The test is already running the function to test that it behaves correctly. The Pointer comparison should simply be deleted.


### Kubernetes version

Happens at HEAD and in v1.27.1.


### Cloud provider

N/A

### OS version

Linux, irrelevant.

### Install tools

N/A

### Container runtime (CRI) and version (if applicable)

N/A

### Related plugins (CNI, CSI, ...) and versions (if applicable)

N/A

--- a/cmd/kubeadm/app/cmd/phases/workflow/runner_test.go
+++ b/cmd/kubeadm/app/cmd/phases/workflow/runner_test.go
@@ -401,14 +401,8 @@ func TestBindToCommandArgRequirements(t *testing.T) {
 					continue
 				}
 
-				// Ensure it is the expected function
-				if reflect.ValueOf(cCmd.Args).Pointer() != reflect.ValueOf(args.args).Pointer() {
-					t.Error("The function pointers where not equal.")
-				}
-
 				// Test passing argument set
 				err := cCmd.Args(cCmd, args.pass)
-
 				if err != nil {
 					t.Errorf("command %s should validate the args: %v\n %v", cCmd.Name(), args.pass, err)
 				}

